src/EntityFramework/Infrastructure/DependencyResolution/DefaultProviderFactoryResolver.cs

```diff
@@ -31,6 +32,11 @@ private static object GetService(Type type, object key, Func<ArgumentException,
                 }
                 catch (ArgumentException e)
                 {
+                    if (string.Equals(name, "System.Data.SqlClient", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return SqlClientFactory.Instance;
```

Comment:
@ajcvickers @divega If we don't find a factory for SqlClient, we'll return this hard-coded one.

Reply:
I think this is fine. It should fix the usability issues, but I am not sure about the full implications. Maybe check it in but then talk about it a bit in the design meeting. It also has some implications for  https://github.com/aspnet/EntityFramework6/issues/823 (e.g. using Microsoft.Data.SqlClient).

Reply:
Agreed that this is fine but we need to talk about #823. I have some ideas.